oauth-1.0a.js

```diff
@@ -48,6 +48,37 @@ function OAuth(opts) {
     this.body_hash_function = opts.body_hash_function || this.hash_function;
 }
 
+// Static methods and properties.
+Object.defineProperties(OAuth, {
+
+    // cache the url regexp.
+    urlPattern: {
+        value: /^(https?):\/\/([^:]+:[^@]+@)?([^:/?#]+)(\:\d+)?(\/[^?#]*)?(\?[^#]*)?(#.*)?$/
+    },
+
+    // Parse url into components.
+    parseUrl: {
```

Comment:
I define the helper as a static method (easier for testing) but I am happy to define it on the prototype like the other helpers if you prefer.

Reply:
is static method is supported by older nodejs or browser?

Reply:
IE9+, Safari 5+ and evergreen browser support `Object.defineProperties`. I am guessing every node versions support ES 5.1; node 0.10 certainly supports `Object.defineProperties`.

If you need better support, I can define it as  `OAuth.parseUrl = function(url) {...}`.

Reply:
how about ``prototype``?